chainer/functions/array/repeat.py

```diff
@@ -0,0 +1,128 @@
+import six
+
+from chainer import cuda
+from chainer import function_node
+from chainer.utils import type_check
+
+
+class Repeat(function_node.FunctionNode):
+
+    """Repeat elements of an array."""
+
+    def __init__(self, repeats, axis=None):
+        if isinstance(repeats, six.integer_types):
+            self.repeats = (repeats,)
+        elif isinstance(repeats, tuple) and all(
+                isinstance(x, six.integer_types) for x in repeats):
+            self.repeats = repeats
+        else:
+            raise TypeError('repeats must be int or tuple of ints')
+
+        if not all(x >= 0 for x in self.repeats):
+            raise ValueError('all elements in repeats must be zero or larger')
+
+        self.axis = axis
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 1)
+
+    def forward(self, inputs):
+        self.retain_inputs((0,))
+        x, = inputs
+        xp = cuda.get_array_module(x)
+        repeats = self.repeats
+        if self.axis is None or len(self.repeats) == 1:
+            repeats = self.repeats[0]
```

Comment:
Only the first element of `repeats` is used if `axis` is `None`.
That seems to be different from `numpy.repeat`'s behavior.

Reply:
Sorry, I should have put blocked-by-another-pr label.
During the implementation of this function, I found an issue that cupy & numpy repeat function behaves differently. cupy/cupy#670
This code cannot be fixed until the above PR is merged.

Reply:
OK.

Reply:
I fixed the code assuming https://github.com/cupy/cupy/pull/670 got merged.